sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -2294,6 +2294,47 @@ public void processElement(
           .containsInAnyOrder(Lists.newArrayList(12, 42, 84, 97), Lists.newArrayList(0, 1, 2));
       pipeline.run();
     }
+
+    @Test
+    @Category({ValidatesRunner.class, UsesStatefulParDo.class, UsesSideInputs.class})
+    public void testStateSideInput() {
+      final PCollectionView<Integer> sideInput =
+          pipeline
+              .apply("CreateSideInput1", Create.of(2))
+              .apply("ViewSideInput1", View.asSingleton());
+
+      TestSimpleStatefulDoFn fn = new TestSimpleStatefulDoFn(sideInput);
+      pipeline.apply(Create.of(KV.of(1, 2))).apply(ParDo.of(fn).withSideInputs(sideInput));
+
+      pipeline.run();
+    }
+
+    private static class TestSimpleStatefulDoFn extends DoFn<KV<Integer, Integer>, Integer> {
+
+      private final PCollectionView<Integer> view;
+
+      @StateId("foo")
+      private final StateSpec<ValueState<Integer>> spec = StateSpecs.value(VarIntCoder.of());
+
+      private TestSimpleStatefulDoFn(PCollectionView<Integer> view) {
+        this.view = view;
+      }
+
+      @ProcessElement
+      public void processElem(ProcessContext c) {
+        // noop
```

Comment:
can you add code here to access the state and the side input?

Reply:
Done.  Somehow python precommit is failing, not related to this tests though I believe